graphql/me/mutation.ts

```diff
@@ -311,8 +311,10 @@ export class MutateMeResolver {
             if (pupil) {
                 throw new PrerequisiteError(`Tried to update pupil data on student`);
             }
+            student.firstname = undefined; // don't allow students to change their names by themselves as their identity is verified during screening
+            student.lastname = undefined;
 
-            await updateStudent(context, prevStudent, { firstname, lastname, ...student });
+            await updateStudent(context, prevStudent, student);
```

Comment:
I'd prefer if we do this check in the updateStudent function instead, then we also cover the updateStudent query. Just as we do in there with email